### ISSUE TITLE
Broken "Handle bins" endpoint

### ISSUE THREAD

**Issue by karafecho** [ExposuresProvider/icees-api#323]
This issue is to report that the "Handle bins" endpoint is returning null results for the ICEES PCD instance. This is a high-priority issue, as it is holding up Brenna's internship work.

**Comment by hyi** [comment]
I cannot reproduce this issue for some reason. The endpoint returns valid results for me with different inputs. I'll need more details in order to reproduce the issue.

**Comment by karafecho** [comment]
The issue surfaced when specifying year 2018. Brenna was focusing on that year for her queries. She was seeing data for both AvgDailyPM2.5Exposure_2 and MaxDailyOzoneExposure_2. However, I just ran the queries below and am not finding any data. I have not yet checked the underlying tables, but we should have airborne pollutant data for year 2018.

```
curl -X 'POST' \
  'https://icees-pcd.renci.org/patient/cohort/COHORT%3A1/feature_association?year=2018' \
  -H 'accept: text/tabular' \
  -H 'Content-Type: application/json' \
  -d '{
  "feature_a": {
    "AvgDailyPM2.5Exposure_2": {
      "operator": "=",
      "value": "1"
    }
  },
  "feature_b": {
    "Sex2": {
      "operator": "=",
      "value": "Male"
    }
  }
}'

```
```
curl -X 'GET' \
  'https://icees-pcd.renci.org/bins?table=patient&feature=AvgDailyPM2.5Exposure_2' \
  -H 'accept: application/json'
```


![image](https://github.com/user-attachments/assets/19b9d34c-3430-4b93-9a14-ab8c0e024d93)

**Comment by hyi** [comment]
I looked into the cmaq data source and found ```AvgDailyPM2.5Exposure_2``` for year 2018 are all empty in cmaq data, yet it has values for years 2017 and 2019, so the root cause of this issue appears to come from the new cmaq data for year 2018. We will need to check with @frostyfan109 to know more about the reason why ```AvgDailyPM2.5Exposure_2``` for year 2018 are all empty.

**Comment by hyi** [comment]
@frostyfan109 Never mind. The new cmap data for year 2018 does include pm25_daily_average data. I am looking into FHIR PIT code to see why AvgDailyPM2.5Exposure_2 for year 2018 are all empty, but the original cmaq data looks correct.

**Comment by hyi** [comment]
@karafecho I believe I find the root cause of the issue. For year 2018, the cmaq pm25 daily average data has a row with nan value for FIPS column, which resulted in FIPS column being read in as float type with each integer FIPS value converted to a float value, e.g., 01001020100 is converted to 01001020100.0. These float FIPS values resulted in empty cmaq values output by FHIR PIT since the FIPS column is used to do inner join with the FIPS column with integer type in geoids.csv to link patients with cmaq data by geographic locations. I have updated my cmaq data merging script to handle nan/null values in cmaq data properly and to convert the FIPS column to integer type and run the updated script for year 2018 to generate the updated merged cmaq data for year 2018 to feed into FHIR PIT. I am currently rerunning FHIR PIT with updated cmaq data for PCD. Once it is done, I'll update ICEES PCD instance with updated database which should fix this issue.

**Comment by frostyfan109** [comment]
@hyi null FIPS values are coming from the CMAQ API, or from elsewhere? Sounds like an issue that needs to be fixed if it's happening on the API-end.

**Comment by hyi** [comment]
@frostyfan109 null FIPS values are only present for year 2018 in 2018_pm25_daily_average.txt file, not present for years 2017 and 2019. Just took a quick look at the data again and verified there is only one row with null FIPS value as shown below:
```
26283613  2018/12/30           NaN        NaN       NaN                        NaN                               NaN
```
As you can see, all columns except for the first Date column are NaN. This row is the last row in the file. Not sure if it is the CMAQ API issue or the data generator script that pulls data using CMAQ API since only the last row has all NaN values. Yes, agreed it sounds like an issue that needs to be fixed on the API-end or the CMAQ data generator script.

**Comment by hyi** [comment]
@karafecho FHIR PIT run with updated cmaq 2018 input data is done and I regenerated database and redeployed PCD dev instance. Now handle bins endpoint returns valid bins data for AvgDailyPM2.5Exposure_2 feature in year 2018. Please check out the PCD dev instance when you get a chance and let me know whether everything looks good. If everything is good, I'll then redeploy PCD prod instance and create a PR to get the updated cmaq merging script merged into FHIR PIT repo.

**Comment by karafecho** [comment]
Just tested the ICEES PCD DEV instance. Looks good. Please move forward with redeployment and updates to the CMAQ merging script.

**Comment by hyi** [comment]
@karafecho Have redployed ICEES PCD PROD instance as well. I tested it out and verified now the handle bins endpoint returns data for 2018. Feel free to test things out and let me know if you see any issues. I have also created the PR [here](https://github.com/ExposuresProvider/FHIR-PIT/pull/32) to get updated CMAQ merge script merged in FHIR PIT repo. I asked you and @frostyfan109 to code review before I can merge it.
